packages/dx-grid-core/src/plugins/virtual-table/helpers.ts

```diff
@@ -60,3 +66,26 @@ export const getViewport: GetViewportFn = (
 
   return result;
 };
+
+export const checkColumnWidths: TableColumnConverterFn = (tableColumns) => {
+  return tableColumns.reduce((acc, tableColumn) => {
+    const { width } = tableColumn;
+    if (typeof width === 'string') {
+      const numb = parseInt(width, 10);
+      const unit = numb ? width.substr(numb.toString().length) : width;
+      const isValidValue = VALID_UNITS.findIndex(validUnit => validUnit === unit) >= 0
+        && !isNaN(numb);
+      if (!isValidValue) {
+        throw new Error(INVALID_TYPE.replace('$1', tableColumn.column!.name));
+      }
+      if (unit === 'px' || unit === '') {
+        acc.push({ ...tableColumn, width: numb });
+      } else {
+        acc.push(tableColumn);
+      }
+    } else {
```

Comment:
A lot of `if/else` operations.

Reply:
Without additional `if/else` we do additional calculations

Reply:
I think without a lot of `if/else` constructions this cod would be cleary.

Also, see this file - plugins/table-column-resizing/computeds.ts

@ushkal can you help us? Should we use a complex `if/else` constructions?